src/index.ts

```diff
@@ -79,31 +84,95 @@ async function setDebugSession(
   app.commands.notifyCommandChanged();
 }
 
+function updateToolbar(
+  widget: NotebookPanel | ConsolePanel | FileEditor,
+  debug: IDebugger
+): void {
+  if (!debug.isDebuggingEnabled) {
+    return;
+  }
+  const isConsolePanel = widget instanceof ConsolePanel;
+
+  const changeAtribute = () => {
+    if (debug.session.debuggedClients.has(debug.session.client.name)) {
+      widget.node.setAttribute('data-jp-debugger', 'true');
+    } else {
+      widget.node.removeAttribute('data-jp-debugger');
+    }
+  };
+
+  const getToolbar = (): Toolbar => {
+    if (isConsolePanel) {
+      return (
+        ((widget as ConsolePanel).widgets.find(
+          widget => widget instanceof Toolbar
+        ) as Toolbar) || new Toolbar()
+      );
+    } else {
+      return (widget as NotebookPanel)?.toolbar;
+    }
+  };
+
+  const insertItemAndCheckIfExist = (toolbar: Toolbar) => {
+    return toolbar.addItem(
+      'debugger-lifeCycle-button',
+      new ToolbarButton({
+        className: 'jp-debugger-switch-button',
+        iconClassName: 'jp-toggle-switch',
+        onClick: async () => {
+          if (
+            !debug.session.debuggedClients.delete(debug.session.client.name)
+          ) {
+            debug.session.debuggedClients.add(debug.session.client.name);
+            await debug.start();
+          } else {
+            await debug.stop();
+          }
+          changeAtribute();
+        },
+        tooltip: 'Enable / Disable Debugger'
+      })
+    );
+  };
+
+  const toolbar = getToolbar();
+
+  if (insertItemAndCheckIfExist(toolbar) && isConsolePanel) {
+    (widget as ConsolePanel).insertWidget(0, toolbar);
+  }
+  changeAtribute();
+}
+
 class DebuggerHandler<
   H extends ConsoleHandler | NotebookHandler | FileHandler
 > {
   constructor(builder: new (option: any) => H) {
     this.builder = builder;
   }
 
-  update<W extends ConsolePanel | NotebookPanel | FileEditor>(
+  async update<W extends ConsolePanel | NotebookPanel | FileEditor>(
     debug: IDebugger,
     widget: W
-  ): void {
+  ): Promise<void> {
     if (!debug.model || this.handlers[widget.id] || !debug.isDebuggingEnabled) {
+      if (!debug.session.debuggedClients.has(debug.session.client.name)) {
+        // in setDebugSession allways we start session to get codeId, if debugging toggle is off we now stop session
+        await debug.stop();
```

Comment:
This could just be `return debug.stop();` instead of awaiting and returning below.

Reply:
(And you you as well remove the comment.)